src/script/team/TeamRepository.ts

```diff
@@ -399,15 +414,17 @@ export class TeamRepository {
   };
 
   private readonly onFeatureConfigUpdate = async (
-    _event: TeamEvent & {name: FEATURE_KEY},
+    event: TeamFeatureConfigurationUpdateEvent,
     source: EventSource,
   ): Promise<void> => {
     if (source !== EventSource.WEBSOCKET) {
       // Ignore notification stream events
       return;
     }
+
     // When we receive a `feature-config.update` event, we will refetch the entire feature config
-    await this.updateFeatureConfig();
+    const {prevFeatureList} = await this.updateFeatureConfig();
+    this.emit('featureUpdated', {event, prevFeatureList});
```

Comment:
suggestion: maybe we could just send the previous value rather than the entire feature config, WDYT?

Reply:
Well, I tried to do it but didn't manage to achieve it type-wise. When accessing `prevFeatureList[event.name]` at this point, the type will not be properly narrowed down to e.g. `FeatureMLS` (we don't know what's `event.name` at compile time, it's `FEATURE_KEY` type)

Event data is nicely tied to its name.
![Screenshot 2023-09-27 at 11 25 17](https://github.com/wireapp/wire-webapp/assets/45733298/fff7034e-88a2-4618-9cf4-0512fbfbd4e5)

But we cannot tie specific feature to it by FEATURE_KEY at this point. It's still union of different features.
![Screenshot 2023-09-27 at 11 27 59](https://github.com/wireapp/wire-webapp/assets/45733298/2818de3c-9bef-4a12-b348-2f86ed7e9c09)

Maybe it's possible but i cannot find a way? Any ideas?

Reply:
ah nevermind we can probably change that later, no biggies here 👌